discovery/service.go

```diff
@@ -52,23 +52,26 @@ func (ds *DiscoveryService) handleDiscovery(dm DiscoveryMessage, addr string) {
 		"message":   dm,
 	}).Debug("Peer discovery received a message")
 
-	if dm.Endpoint == ds.c.NodeId {
+	if ds.c.HasEndpoint(dm.Endpoint) {
 		log.WithFields(log.Fields{
 			"discovery": ds,
 			"peer":      addr,
 			"message":   dm,
-		}).Debug("Peer discovery is from this node, dropping")
-
+		}).Debug("Discovery message for one of our own ids, dropping")
 		return
 	}
 
 	var client cla.Convergence
 	switch dm.Type {
-	case MTCP:
+	case core.MTCP:
 		client = mtcp.NewMTCPClient(fmt.Sprintf("%s:%d", addr, dm.Port), dm.Endpoint, false)
+		ds.c.RegisterConvergable(client)
 
-	case TCPCL:
-		client = tcpcl.DialClient(fmt.Sprintf("%s:%d", addr, dm.Port), ds.c.NodeId, false)
+	case core.TCPCL:
+		for _, eid := range ds.c.CLAdapters[core.TCPCL] {
```

Comment:
The map access requires that the key is always present. Even if this is _now_ given, a check might be quite good.